### ISSUE TITLE
setUnhandledExceptionHook hook not called in library

### ISSUE THREAD

**Issue by josephdwyer** [JetBrains/kotlin-native#2941]
This example demonstrates the hook set with `setUnhandledExceptionHook` not being called when the kotlin native library is being consumed from c.

knlibrary.kt
```
package knlibrary

import kotlin.native.*
import kotlin.native.concurrent.*

fun test() {
    println("Hello Kotlin/Native!")

    setUnhandledExceptionHook({
        throwable: Throwable -> OnUnhandledException(throwable)
    }.freeze())

    CausesError()

    println("Should not get here")
}

fun OnUnhandledException(t: Throwable) {
    println("EXCEPTION: ${t.message}")
    t.printStackTrace()
}

fun CausesError() {
    val result = 1 / 0
}
```

consumer.c
```
#include <stdio.h>
#include "knlibrary_api.h"

int main()
{
   printf("Hello, C!\n");

   knlibrary_ExportedSymbols* lib = knlibrary_symbols();
   lib->kotlin.root.knlibrary.test();

   return 0;
}
```

Compile the example:
```
kotlinc-native knlibrary.kt -produce dynamic -o knlibrary.dll
```

Using instructions from [here](https://kotlinlang.org/docs/tutorials/native/dynamic-libraries.html#compiling-and-running-the-example-on-windows) to consume the library.
Note: We have to rename the dll because `consumer.exe` looks for the name with `_symbols` for whatever reason.
```
lib.exe /def:knlibrary_symbols.def /out:knlibrary_symbols.lib
cl.exe consumer.c knlibrary_symbols.lib
MOVE knlibrary.dll knlibrary_symbols.dll
consumer.exe
```

Program output:
```
>consumer.exe
Hello, C!
Hello Kotlin/Native!

```

Expected: output includes stack trace

**Comment by olonho** [comment]
Problem is reproducible only if host executable is compiled with MSVC, if it is compiled with clang - top level handler still catches the exception. https://github.com/JetBrains/kotlin-native/pull/2946 fixes the issue for MSVC hosted executables as well.

**Comment by olonho** [comment]
Fixed in master and will be available in 1.3.40.
